# Community composition

## Gut microbiota

```{r gut_load_data_tax}
load("data/gut/data.Rdata")
```

### Taxonomy overview 

#### Phylum level

```{r taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(~factor(environment, labels=c("low" = "Low altitude", "high" = "High altitude")),  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_blank(),
          axis.ticks.x = element_blank(),
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "white"),
          strip.text = element_text(size = 12, lineheight = 0.6,face="bold"),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

**Number of MAGs**
```{r mag, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nmags <- nrow(genome_counts)
cat(nmags)
```

**Number of bacteria phyla**

```{r phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() %>% 
  cat()
```

**Number of Archaea phyla**

```{r arch, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()%>% 
  cat()
```

***Phylum relative abundances***

```{r taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum,river, environment) %>%
  summarise(relabun=sum(count))
```

```{r taxonomy_phylum_summary_envir, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,Total,High,Low)
```
```{r taxonomy_jitterplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    left_join(genome_metadata %>% select(phylum,phylum) %>% unique(),by=join_by(phylum==phylum)) %>%
#    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    filter(phylum %in% phylum_arrange[1:20]) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~environment)+
        theme_minimal() + 
        labs(y="phylum", x="Relative abundance", color="Phylum")
```

**Bacteria phyla in individuals from low altitude**

```{r phyla_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}

low_samples <- sample_metadata %>% 
  filter(environment=="low") %>% 
  dplyr::select(sample) %>% 
  pull()

low_genomes <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  select(all_of(low_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% low_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() %>% 
  cat()
```


**Bacteria phyla in individuals from high altitude**

```{r phyla_cap, comment="", echo=FALSE, message=FALSE, warning=FALSE}

high_samples <- sample_metadata %>% 
  filter(environment=="high") %>% 
  dplyr::select(sample) %>% 
  pull()

high_genomes <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  select(all_of(high_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% high_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() %>% 
  cat()
```
```{r taxonomy_phylum_summary_river, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              le_mean=mean(relabun[river=="Leitzaran"]*100, na.rm=T),
             le_sd=sd(relabun[river=="Leitzaran"]*100, na.rm=T),
              ha_mean=mean(relabun[river=="Harpea"]*100, na.rm=T),
              ha_sd=sd(relabun[river=="Harpea"]*100, na.rm=T),
              er_mean=mean(relabun[river=="Erlan"]*100, na.rm=T),
              er_sd=sd(relabun[river=="Erlan"]*100, na.rm=T),
              go_mean=mean(relabun[river=="Goizueta"]*100, na.rm=T),
              go_sd=sd(relabun[river=="Goizueta"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Leitzaran=str_c(round(le_mean,3),"±",round(le_sd,3)),
           Harpea=str_c(round(ha_mean,3),"±",round(ha_sd,3)),
           Erlan=str_c(round(er_mean,3),"±",round(er_sd,3)),
           Goizueta=str_c(round(go_mean,3),"±",round(go_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,Total,Leitzaran,Goizueta,Harpea,Erlan)
```

#### Family level

***Percentange of families in each group***

```{r taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family, river,environment) %>%
  summarise(relabun=sum(count))
```

```{r taxonomy_family_summary_origin, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(family,Total,High,Low)
```

```{r taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

# Per environment
family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~environment)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

```{r taxonomy_family_summary_river, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              le_mean=mean(relabun[river=="Leitzaran"]*100, na.rm=T),
             le_sd=sd(relabun[river=="Leitzaran"]*100, na.rm=T),
              ha_mean=mean(relabun[river=="Harpea"]*100, na.rm=T),
              ha_sd=sd(relabun[river=="Harpea"]*100, na.rm=T),
              er_mean=mean(relabun[river=="Erlan"]*100, na.rm=T),
              er_sd=sd(relabun[river=="Erlan"]*100, na.rm=T),
              go_mean=mean(relabun[river=="Goizueta"]*100, na.rm=T),
              go_sd=sd(relabun[river=="Goizueta"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Leitzaran=str_c(round(le_mean,3),"±",round(le_sd,3)),
           Harpea=str_c(round(ha_mean,3),"±",round(ha_sd,3)),
           Erlan=str_c(round(er_mean,3),"±",round(er_sd,3)),
           Goizueta=str_c(round(go_mean,3),"±",round(go_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(family,Total,Leitzaran,Goizueta,Harpea,Erlan)
```

#### Genus level

*** Percetange of genera in each group***
```{r taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,phylum,genus, environment) %>%
  summarise(relabun=sum(count)) 
# %>%
#   filter(genus != "g__") %>%
#   mutate(genus= sub("^g__", "", genus))

genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(genus,Total,High,Low) 
```

```{r taxonomy_jitterplot_genus, fig.height=14, fig.width=10, fig.fullwidth=TRUE}
genus_summary_sort <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=T),sd=sd(relabun, na.rm=T)) %>%
    arrange(-mean) 

genus_summary %>%
  mutate(genus=factor(genus, levels=rev(genus_summary_sort %>% pull(genus)))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
  scale_color_manual(values=phylum_colors) +
  geom_jitter(alpha=0.5) + 
  facet_grid(.~environment)+
  theme_minimal() + 
  theme(axis.text.y = element_text(size=6))+
  labs(y="Family", x="Relative abundance", color="Phylum")

```
**Number of MAGs without genera-level annotation**
```{r nongenera_gut, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(genus == "g__") %>%
  nrow() %>% 
  cat()
```

```{r nongene_phylum_gut, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(genus == "g__") %>%
  group_by(phylum) %>%
  summarize(count_nogene = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>% 
  mutate(percentage=100*count_nogene/count_total) %>% 
  tt()
```

**Percentage of MAGs without genus-level annotation**
```{r gen_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nongenera <- genome_metadata %>%
  filter(genus == "g__") %>%
  summarize(Mag_nogenera = n()) %>% 
  pull()
perct <- nongenera*100/nmags
cat(perct)
```


**Number of MAGs without species-level annotation**
```{r nonspe, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n())

```
```{r nonspe_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(species == "s__") %>%
  group_by(phylum) %>%
  summarize(count_nospecies = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>%
  mutate(species_annotated=count_total-count_nospecies) %>% 
  mutate(percentage=100*count_nospecies/count_total) %>% 
  mutate(percentage_species=100-100*count_nospecies/count_total)%>% 
  tt()
```

**Percentage of MAGs without species-level annotation**
```{r sp_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nonspecies <- genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n()) %>% 
  pull()
perct <- nonspecies*100/nmags
cat(perct)
```



## Skin microbiota

```{r skin_load_data_tax}
load("data/skin/data.Rdata")
```

### Taxonomy overview 

#### Phylum level

```{r skin_taxonomy_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=phylum, group=phylum)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_manual(values=phylum_colors) +
    facet_nested(~factor(environment, labels=c("low" = "Low altitude", "high" = "High altitude")),  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.ticks.x = element_blank(),
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "white"),
          strip.text = element_text(size = 12, lineheight = 0.6,face="bold"),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Phylum",y = "Relative abundance",x="Samples")
```

**Number of MAGs**
```{r skin_mag, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nmags <- nrow(genome_counts)
cat(nmags)
```

**Number of bacteria phyla**

```{r skin_phyla, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() %>% 
  cat()
```

**Number of Archaea phyla**

```{r skin_arch, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>% 
  filter(domain == "d__Archaea")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length()%>% 
  cat()
```

***Phylum relative abundances***

```{r skin_taxonomy_phylum_summary, warning=FALSE, comments="", message=FALSE}
phylum_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>%
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>%
  left_join(sample_metadata, by = join_by(sample == sample)) %>%
  left_join(genome_metadata, by = join_by(genome == genome)) %>%
  group_by(sample,phylum, environment, river) %>%
  summarise(relabun=sum(count))
```

```{r skin_taxonomy_phylum_summary_skin, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,6),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,Total,High,Low)
```
```{r skin_taxonomy_jitterplot_phylum, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
phylum_arrange <- phylum_summary %>%
    group_by(phylum) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(phylum) %>%
    pull()

phylum_summary %>%
    left_join(genome_metadata %>% select(phylum,phylum) %>% unique(),by=join_by(phylum==phylum)) %>%
#    left_join(sample_metadata,by=join_by(sample==sample)) %>%
    filter(phylum %in% phylum_arrange[1:20]) %>%
    mutate(phylum=factor(phylum,levels=rev(phylum_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=phylum, group=phylum, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~environment)+
        theme_minimal() + 
        labs(y="phylum", x="Relative abundance", color="Phylum")
```

**Bacteria phyla in individuals from low altitude**

```{r skin_phyla_nat, comment="", echo=FALSE, message=FALSE, warning=FALSE}

low_samples <- sample_metadata %>% 
  filter(environment=="low") %>% 
  dplyr::select(sample) %>% 
  pull()

low_genomes <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  select(all_of(low_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% low_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```


**Bacteria phyla in individuals from high altitude**

```{r skin_phyla_cap, comment="", echo=FALSE, message=FALSE, warning=FALSE}

high_samples <- sample_metadata %>% 
  filter(environment=="high") %>% 
  dplyr::select(sample) %>% 
  pull()

high_genomes <- genome_counts_filt %>% 
  column_to_rownames("genome") %>% 
  select(all_of(high_samples)) %>%
  as.data.frame() %>%
  filter(rowSums(across(where(is.numeric)))!=0)%>% 
  rownames_to_column("genome")%>% 
  dplyr::select(genome) %>% 
  pull()

genome_metadata %>% 
  filter(genome %in% high_genomes) %>% 
  filter(domain == "d__Bacteria")%>%
  dplyr::select(phylum) %>%
  unique() %>%
  pull() %>%
  length() 
```

```{r skin_taxonomy_phylum_summary_river, warning=FALSE, comments="", message=FALSE}
phylum_summary %>%
    group_by(phylum) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              le_mean=mean(relabun[river=="Leitzaran"]*100, na.rm=T),
             le_sd=sd(relabun[river=="Leitzaran"]*100, na.rm=T),
              ha_mean=mean(relabun[river=="Harpea"]*100, na.rm=T),
              ha_sd=sd(relabun[river=="Harpea"]*100, na.rm=T),
              er_mean=mean(relabun[river=="Erlan"]*100, na.rm=T),
              er_sd=sd(relabun[river=="Erlan"]*100, na.rm=T),
              go_mean=mean(relabun[river=="Goizueta"]*100, na.rm=T),
              go_sd=sd(relabun[river=="Goizueta"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Leitzaran=str_c(round(le_mean,6),"±",round(le_sd,3)),
           Harpea=str_c(round(ha_mean,6),"±",round(ha_sd,3)),
           Erlan=str_c(round(er_mean,6),"±",round(er_sd,3)),
           Goizueta=str_c(round(go_mean,6),"±",round(go_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(phylum,Total,Leitzaran,Goizueta,Harpea,Erlan)
```

#### Class level

***Percentange of classes in each group***

```{r skin_taxonomy_class_summary, warning=FALSE, comments="", message=FALSE}
class_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,class, river,environment) %>%
  summarise(relabun=sum(count))
```

```{r skin_taxonomy_class_summary_origin, warning=FALSE, comments="", message=FALSE}
class_summary %>%
    group_by(class) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(class,Total,High,Low)
```

#### Family level

***Percentange of families in each group***

```{r skin_taxonomy_family_summary, warning=FALSE, comments="", message=FALSE}
family_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,family, river,environment) %>%
  summarise(relabun=sum(count))
```

```{r skin_taxonomy_family_summary_origin, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(family,Total,High,Low)
```

```{r skin_taxonomy_jitterplot_family, warning=FALSE, comments="", message=FALSE, fig.height=8, fig.width=10, fig.fullwidth=TRUE}
family_arrange <- family_summary %>%
    group_by(family) %>%
    summarise(mean=sum(relabun)) %>%
    arrange(-mean) %>%
    select(family) %>%
    pull()

# Per environment
family_summary %>%
    left_join(genome_metadata %>% select(family,phylum) %>% unique(),by=join_by(family==family)) %>%
    filter(family %in% family_arrange[1:20]) %>%
    mutate(family=factor(family,levels=rev(family_arrange[1:20]))) %>%
    filter(relabun > 0) %>%
    ggplot(aes(x=relabun, y=family, group=family, color=phylum)) +
        scale_color_manual(values=phylum_colors[-8]) +
        geom_jitter(alpha=0.5) + 
        facet_grid(.~environment)+
        theme_minimal() + 
        labs(y="Family", x="Relative abundance", color="Phylum")

```

```{r skin_taxonomy_family_summary_river, warning=FALSE, comments="", message=FALSE}
family_summary %>%
    group_by(family) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              le_mean=mean(relabun[river=="Leitzaran"]*100, na.rm=T),
             le_sd=sd(relabun[river=="Leitzaran"]*100, na.rm=T),
              ha_mean=mean(relabun[river=="Harpea"]*100, na.rm=T),
              ha_sd=sd(relabun[river=="Harpea"]*100, na.rm=T),
              er_mean=mean(relabun[river=="Erlan"]*100, na.rm=T),
              er_sd=sd(relabun[river=="Erlan"]*100, na.rm=T),
              go_mean=mean(relabun[river=="Goizueta"]*100, na.rm=T),
              go_sd=sd(relabun[river=="Goizueta"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           Leitzaran=str_c(round(le_mean,3),"±",round(le_sd,3)),
           Harpea=str_c(round(ha_mean,3),"±",round(ha_sd,3)),
           Erlan=str_c(round(er_mean,3),"±",round(er_sd,3)),
           Goizueta=str_c(round(go_mean,3),"±",round(go_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(family,Total,Leitzaran,Goizueta,Harpea,Erlan)
```

```{r skin_taxonomy_fam_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=family, group=family)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    scale_fill_viridis(discrete = TRUE)+
    facet_nested(~factor(environment, labels=c("low" = "Low altitude", "high" = "High altitude")),  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.ticks.x = element_blank(),
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "white"),
          strip.text = element_text(size = 12, lineheight = 0.6,face="bold"),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Family",y = "Relative abundance",x="Samples")
```
***Family***
```{r phyloseq_skin, comment="", echo=FALSE, message=FALSE, warning=FALSE}
#Phyloseq object
count_phy <- genome_counts_filt %>%
  column_to_rownames(var="genome")%>%
  otu_table(., taxa_are_rows=T)

sample_info_tab_phy <- sample_metadata%>%
  column_to_rownames(var="sample")%>%
  sample_data()

TAX <- genome_metadata%>%
  column_to_rownames(var="genome")%>%
  select(1:7)%>%
  as.matrix()%>%
  tax_table()
tree <- phy_tree(genome_tree)

physeq_all = phyloseq(count_phy, TAX, sample_info_tab_phy, tree)
physeq_all_clr <- microbiome::transform(physeq_all, 'clr')

```
```{r fam_MW, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_clr_family <- microbiome::aggregate_taxa(physeq_all_clr, 'family')
clr_t_family <- as.data.frame(t(as.matrix(physeq_clr_family@otu_table))) %>%
  rownames_to_column(., "sample")%>% 
  left_join(., sample_metadata[c(1,10)], by = join_by(sample == sample))
significant_family <- clr_t_family %>%
    pivot_longer(-c(sample,environment), names_to = "Family", values_to = "value") %>%
    group_by(Family) %>%
    summarise(p_value = wilcox.test(value ~ environment, exact = FALSE, alternative = "less")$p.value) %>%
    mutate(p_adjust=p.adjust(p_value, method="BH")) %>%
    filter(p_adjust < 0.05)
significant_family
```

```{r family_plot, comment="", echo=FALSE, message=FALSE, warning=FALSE}
physeq_family <- microbiome::aggregate_taxa(physeq_all, 'family')
physeq_family_rel <-  microbiome::transform(physeq_family, "compositional")
family <- as.data.frame(t(as.matrix(physeq_family_rel@otu_table)))
significant_family_table <- family[, colnames(family) %in% significant_family$Family, drop=FALSE] %>%
  rownames_to_column(., "sample")%>%  
  left_join(., sample_metadata[c(1,10)], by = join_by(sample == sample))%>%
  select(-sample)

ggplot(significant_family_table, aes(x=environment, y=f__Methylophilaceae, color = environment, fill=environment)) +
    geom_boxplot(alpha = 0.2, outlier.shape = NA, width = 0.3, show.legend = FALSE) +
  geom_jitter(width = 0.1, show.legend = TRUE) +
  theme_minimal() +
  theme(
    axis.line = element_line(size = 0.5, linetype = "solid", colour = "black"),
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    panel.border = element_blank())
```

#### Genus level

```{r skin_taxonomy_genus_summary, warning=FALSE, comments="", message=FALSE}
genus_summary <- genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  left_join(genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  group_by(sample,phylum,genus, environment) %>%
  summarise(relabun=sum(count)) 
# %>%
#   filter(genus != "g__") %>%
#   mutate(genus= sub("^g__", "", genus))

genus_summary %>%
    group_by(genus) %>%
    summarise(total_mean=mean(relabun*100, na.rm=T),
              total_sd=sd(relabun*100, na.rm=T),
              High_mean=mean(relabun[environment=="high"]*100, na.rm=T),
             High_sd=sd(relabun[environment=="high"]*100, na.rm=T),
              Low_mean=mean(relabun[environment=="low"]*100, na.rm=T),
              Low_sd=sd(relabun[environment=="low"]*100, na.rm=T)) %>%
    mutate(Total=str_c(round(total_mean,3),"±",round(total_sd,3)),
           High=str_c(round(High_mean,3),"±",round(High_sd,3)),
           Low=str_c(round(Low_mean,3),"±",round(Low_sd,3))) %>% 
    arrange(-total_mean) %>% 
    dplyr::select(genus,Total,High,Low) %>% 
  tt()
```

```{r skin_taxonomy_jitterplot_genus, fig.height=14, fig.width=10, fig.fullwidth=TRUE}
genus_summary_sort <- genus_summary %>%
    group_by(genus) %>%
    summarise(mean=mean(relabun, na.rm=T),sd=sd(relabun, na.rm=T)) %>%
    arrange(-mean) 

genus_summary %>%
  mutate(genus=factor(genus, levels=rev(genus_summary_sort %>% pull(genus)))) %>%
  filter(relabun > 0) %>%
  ggplot(aes(x=relabun, y=genus, group=genus, color=phylum)) +
  scale_color_manual(values=phylum_colors) +
  geom_jitter(alpha=0.5) + 
  facet_grid(.~environment)+
  theme_minimal() + 
  theme(axis.text.y = element_text(size=6))+
  labs(y="Family", x="Relative abundance", color="Phylum")

```

```{r skin_taxonomy_gen_barplot, fig.height=6, fig.width=10, fig.fullwidth=TRUE}
genome_counts_filt %>%
  mutate_at(vars(-genome),~./sum(.)) %>% #apply TSS nornalisation
  pivot_longer(-genome, names_to = "sample", values_to = "count") %>% #reduce to minimum number of columns
  left_join(., genome_metadata, by = join_by(genome == genome)) %>% #append genome metadata
  left_join(., sample_metadata, by = join_by(sample == sample)) %>% #append sample metadata
  filter(count > 0) %>% #filter 0 counts
  ggplot(., aes(x=sample,y=count, fill=genus, group=genus)) + #grouping enables keeping the same sorting of taxonomic units
    geom_bar(stat="identity", colour="white", linewidth=0.1) + #plot stacked bars with white borders
    facet_nested(~factor(environment, labels=c("low" = "Low altitude", "high" = "High altitude")),  scales="free") + #facet per day and treatment
    guides(fill = guide_legend(ncol = 1)) +
    theme(axis.text.x = element_text(angle = 45, hjust = 1),
          axis.ticks.x = element_blank(),
          axis.title.x = element_blank(),
          panel.background = element_blank(),
          panel.border = element_blank(),
          panel.grid.major = element_blank(),
          panel.grid.minor = element_blank(),
          strip.background = element_rect(fill = "white"),
          strip.text = element_text(size = 12, lineheight = 0.6,face="bold"),
          axis.line = element_line(linewidth = 0.5, linetype = "solid", colour = "black")) +
   labs(fill="Genera",y = "Relative abundance",x="Samples")
```

**Number of MAGs without genera-level annotation**
```{r skin_nongenera, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(genus == "g__") %>%
  nrow() %>% 
  cat()
```

```{r skin_nongene_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(genus == "g__") %>%
  group_by(phylum) %>%
  summarize(count_nogene = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>% 
  mutate(percentage=100*count_nogene/count_total) %>% 
  tt()
```

**Percentage of MAGs without genus-level annotation**
```{r skin_gen_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nongenera <- genome_metadata %>%
  filter(genus == "g__") %>%
  summarize(Mag_nogenera = n()) %>% 
  pull()
perct <- nongenera*100/nmags
cat(perct)
```


**Number of MAGs without species-level annotation**
```{r skin_nonspe, comment="", echo=FALSE, message=FALSE, warning=FALSE}
genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n())

```
```{r skin_nonspe_phylum, comment="", echo=FALSE, message=FALSE, warning=FALSE}
total_mag_phylum <- genome_metadata %>%
  group_by(phylum) %>%
  summarize(count_total = n())
genome_metadata %>%
  filter(species == "s__") %>%
  group_by(phylum) %>%
  summarize(count_nospecies = n()) %>% 
  left_join(total_mag_phylum, by = join_by(phylum == phylum)) %>% 
  mutate(species_annotated=count_total-count_nospecies) %>% 
  mutate(percentage_non_species=100*count_nospecies/count_total) %>% 
  mutate(percentage_species=100-100*count_nospecies/count_total)%>% 
  tt()
```

**Percentage of MAGs without species-level annotation**
```{r skin_sp_percet, comment="", echo=FALSE, message=FALSE, warning=FALSE}
nonspecies <- genome_metadata %>%
  filter(species == "s__") %>%
  summarize(Mag_nospecies = n()) %>% 
  pull()
perct <- nonspecies*100/nmags
cat(perct)
```




